server/rosetta/cosmos/conversion/data.go

```diff
@@ -0,0 +1,297 @@
+	return &types.BlockIdentifier{
+		Index: block.Block.Height,
+		Hash:  block.Block.Hash().String(),
+	}
+}
+
+func SdkMsgsToOperations(msgs []sdk.Msg, withStatus bool, feeLen int) []*types.Operation {
+	var operations []*types.Operation
+	var status string
+	for i, msg := range msgs {
+		switch msg.Type() { // nolint
+		case rosetta.OperationSend:
+			newMsg := msg.(*banktypes.MsgSend)
+			fromAddress := newMsg.FromAddress
+			toAddress := newMsg.ToAddress
+			amounts := newMsg.Amount
+			if len(amounts) == 0 {
+				continue
+			}
+			coin := amounts[0]
+			sendOp := func(account, amount string, index int) *types.Operation {
+				if withStatus {
+					status = rosetta.StatusSuccess
+				}
+				return &types.Operation{
+					OperationIdentifier: &types.OperationIdentifier{
+						Index: int64(index),
+					},
+					Type:   rosetta.OperationSend,
+					Status: status,
+					Account: &types.AccountIdentifier{
+						Address: account,
+					},
+					Amount: &types.Amount{
+						Value: amount,
+						Currency: &types.Currency{
+							Symbol: coin.Denom,
+						},
+					},
+				}
+			}
+			operations = append(operations,
+				sendOp(fromAddress, "-"+coin.Amount.String(), feeLen+i),
+				sendOp(toAddress, coin.Amount.String(), feeLen+i+1),
+			)
+		}
+	}
+	return operations
+}
+
+func GetMsgsFromOperations(ops []*types.Operation) (sdk.Msg, sdk.Coins, error) {
+	var feeAmnt []*types.Amount
+	var sendOps []*types.Operation
+	if len(ops) == 2 {
+		sendMsg, err := GetTransferTxDataFromOperations(ops)
+		return sendMsg, nil, err
+	} else if len(ops) == 3 {
+		for _, op := range ops {
+			if op.Type == rosetta.OperationFee {
+				amount := op.Amount
+				feeAmnt = append(feeAmnt, amount)
+			}
+			if op.Type == rosetta.OperationSend {
+				sendOps = append(sendOps, op)
+			}
+		}
+	}
+	sendMsg, err := GetTransferTxDataFromOperations(sendOps)
+	if err != nil {
+		return nil, nil, err
+	}
+	return sendMsg, ConvertAmountToCoins(feeAmnt), nil
+}
+
+func ConvertAmountToCoins(amounts []*types.Amount) sdk.Coins {
+	var feeCoins sdk.Coins
+	for _, amount := range amounts {
+		absValue := strings.Trim(amount.Value, "-")
+		value, err := strconv.ParseInt(absValue, 10, 64)
+		if err != nil {
+			return nil
+		}
+		coin := sdk.NewCoin(amount.Currency.Symbol, sdk.NewInt(value))
+		feeCoins = append(feeCoins, coin)
+	}
+	return feeCoins
+}
+
+// GetTransferTxDataFromOperations extracts the from and to addresses from a list of operations.
+// We assume that it comes formated in the correct way. And that the balance of the sender is the same
+// as the receiver operations.
+func GetTransferTxDataFromOperations(ops []*types.Operation) (*banktypes.MsgSend, error) {
+	var (
+		from, to sdk.AccAddress
+		sendAmt  sdk.Coin
+		err      error
+	)
+
+	for _, op := range ops {
+		if strings.HasPrefix(op.Amount.Value, "-") {
+			from, err = sdk.AccAddressFromBech32(op.Account.Address)
+			if err != nil {
+				return nil, err
+			}
+		} else {
```

Comment:
```suggestion

		continue
		}
```

Then you can reduce intendation of the now removed `else` block (improves readability).